---
title: DETERMINING GENETIC FACTORS ASSOCIATED WITH HIGH BLOOD PRESSURE
author: Aulia Khoirunnisa - 2915263K
number-sections: false
fontsize: 12pt
geometry: margin=1in
format:
  html:
    embed-resources: true
    code-tools: true
  pdf: 
    includes:
      in_header: preamble_2915263K.tex
    number-sections: false
    fig-pos: "H"
    toc: true
    toc_depth: 3
    keep_tex: true
    latex_engine: xelatex
header-includes:
  - \usepackage{setspace}
  - \singlespacing
editor_options: 
  chunk_output_type: console
execute:
  echo: false
  eval: true
  warning: false
  message: false
bibliography: References_2915263K.bib
bibliographystyle: apacite
---

```{=tex}
% Cover Page
\begin{titlepage}
  \centering
  \vspace*{2cm}
  \Huge
  \textbf{DETERMINING GENETIC FACTORS ASSOCIATED WITH HIGH BLOOD PRESSURE}\\
  \vspace{1.5cm}
  \Large
  \textbf{Aulia Khoirunnisa (2915263K)}
  \vfill
  \textbf{August 2024}
  \vspace{0.8cm}
\end{titlepage}
\clearpage

% Pages with Roman Numbering
\pagenumbering{roman}
\setcounter{page}{1}

% Abstract Section
\section*{ABSTRACT}
\addcontentsline{toc}{section}{ABSTRACT}

Background: Research on the causes of high blood pressure has rapidly advanced, focusing on both clinical and genetic components as the associated risks continue to increase. Particularly, Genome-Wide Association Studies (GWAS), a more extensive research to determine specific genetics markers such as Single Nucleotide Polymorphisms (SNPs). Then, this study conducted to examine the relationship between the clinical measurements represented as age, sex, body mas index and smoking status as well as genetic factors SNPs to blood pressure traits both on systolic and diastolic. 

Methodology: The analysis was applied using data from Glasgow Medical Centre, consisting of two datasets - blood pressure and SNPs, with missing values and collinearity existed. Some variables particularly on SNPs dataset were removed due to high correlation to improve the model's efficiency. All variables kept preserved and applied random forest approach to fill missing values while maintain the integrity of the dataset.

Analysis of the Data: First, the data was projected in the visualization to discover the relationship with the response variable. Using multivariate linear regression model, it was learnt for the clinical covariates that age, sex and smoking status were significantly correlated to the systolic blood pressure while only age and sex did the same for the diastolic blood pressure. Meanwhile for the genetic factors, only one SNP rs2782980\_T was found to be significant to the systolic blood pressure. On the other hand, there was no SNPs candidate appear to be associated with diastolic blood pressure. 

Discussion: Considering only the significant clinical measurement, the predictors explained the systolic blood pressure for 5.9\% of the variance of sbp. This slightly improved to 6.05\% when combining with the genetic factor as discovered earlier. In contrast, the variation of the diastolic blood pressure was explained by only 1.34\% of the variance. These findings should consider the limitation, as the small and homogenous cohort restrains the statistical power of the study, potentially leading to the non-detection of significant genetic associations with blood pressure.  

Keywords: systolic blood pressure, diastolic blood pressure, genetics factors, variation, multivariate linear regression model, Glasgow Mecial Centre


\clearpage

% Table of Contents, List of Figures, List of Tables
\listoffigures
\clearpage

\listoftables
\newpage

% Switch to Arabic Numbering for Main Content
\pagenumbering{arabic}
\setcounter{page}{1}
```
```{r}
library(tidyverse)
library(gt)
library(dplyr)
library(ggplot2)
library(lmtest)
library(MASS)
library(visdat)
library(plotly)
library(missRanger)
library(corrplot)
library(reshape2)
library(explore)
library(ISLR)
library(skimr)
library(caret)
library(gt)
library(gridExtra)
library(dplyr)
library(ggplot2)
library(e1071)
library(car)
library(glmnet)
library(mgcv)
library(visdat)
```

# INTRODUCTION

## Background

According to [@who_hypertension], hypertension, generally known as high blood pressure, is a critical health issue concerning to more than 1.28 billion adults globally. This condition is categorized by two key measurements. The first case is Systolic Blood Pressure (*sbp*) which measures the pressure in the arteries when the heart beats, and will be determined as high if it consistently reads over 140 mmHg on multiple occasions. Then, Diastolic Blood Pressure (*dbp*), demonstrates the pressure in the arteries when the heart is at rest between beats, with readings above 90 mmHg being sympthoms of hypertension.

Research into the risk factors for high blood pressure has been emerging which more focusing on both clinical measurements and genetic tendencies. One of them is [@cheng2022age] which highlighted how age influences the risk of hypertension, showing significant changes in both high blood pressures among specific age groups of both men and women. Consequently, a study by [@andr2020] also implied that daily cigarette consumption and smoking status significantly impact the risk of developing the hypertension.

Further advancements in understanding hypertension involve genetic research, particularly Genome-Wide Association Studies (GWAS), which aim to identify specific genetic markers such as the Single Nucleotide Polymorphisms (SNPs) associated with high blood pressure. These studies have led the discovery of numerous genetic loci (specific gene locations) that contribute to high blood pressure variablity. Genetic factors are estimated to account for 20-50% of the risk of developing high blood pressure [@yama2015]. Notably, a study by [@evan2018] also identified 535 new genetic loci linked to blood pressure by analyzing data from over one million individuals.

This study aims to elaborate the previous research by examining the relationship between high blood pressure with both clinical and genetic factors. Specifically, it focuses on the associations between clinical measurements - such as age, sex, body mass index (BMI), and smoking status - along with genetic components represented by Single Nucleotide Polymorphisms (SNPs). SNPs are variations at specific positions in the DNA sequence, where the nucleotide may differ between individuals. In this study, SNPs are quantified based on the number of minor alleles (0, 1, or 2) present at 14 specific genomic locations. This research seeks to understand how these clinical and genetic factors collectively influence blood pressure variation both on systolic and diastolic.

## Research Questions

In this research, I conducted a simplified study related to the factors cause of high blood pressure both in systolic (*sbp*) and diastolic (*dbp*) on a small cohort of patients from the Glasgow Medical Centre. The initial step involved examining the number of clinical measurements represented as *age*, *sex*, body mas index (*bmi*) and smoking status (*newsmoke*) of each individual to determine how these factors explain the variation of blood pressure. Then the next process aimed to detect the genetic components association, portrayed by SNPs candidate, to blood pressure traits. Hence, by combining the significant clinical and genetics factors, this study achieved to discover the extent of blood pressure variation that can be explained by these combined factors.

```{r echo=FALSE}

bloodpressure <- read.csv("/Users/auliakhoirunnisa/Documents/UofG/Capstone Project/Dataset/[2915263] Determining genetic variation associated with heart disease - Determining genetic factors associated with high blood pressure/data/bloodpressure.csv")
snps <- read.csv("/Users/auliakhoirunnisa/Documents/UofG/Capstone Project/Dataset/[2915263] Determining genetic variation associated with heart disease - Determining genetic factors associated with high blood pressure/data/snpdata.csv")
colnames(snps)[colnames(snps) == "SEX"] <- "sex"
```

# METHODOLOGY

## Data Collection

The two datasets, ***bloodpressure*** and ***snps***, comprise both continuous and categorical variables, totaling 18 covariates and 2 response variables. Both datasets have missing data, and some degree of collinearity exists among the measured variables. The response variables in this analysis are systolic blood pressure (*sbp*) and diastolic blood pressure (*dbp*). The total observations recorded on ***bloodpressure*** are 5,262 while the ***snps*** dataset accounts for 3,850. These dataset can be linked using *IID* and *sex;* however not all of observations can be merged due to the absence of genetic data for some individuals. A detailed description of the datasets is provided in the table below:

```{r, include = FALSE}

bloodpressure <- bloodpressure %>%
  mutate(
    age = as.numeric(age),
    sex = as.factor(sex),
    bmi = as.numeric(bmi),
    newsmoke = as.factor(newsmoke),
    sbp = as.numeric(sbp),
    dbp = as.numeric(dbp)
  )
```

```{r}
#| label: tbl-desc-bloodpressure
#| tbl-cap: The Blood Pressure Dataset Overview
#| tbl-align: center

variable_bp <- colnames(bloodpressure)
description_bp <- c("Unique code for each individual", 
                 "Age of each individual", 
                 "1: male, 2: female", 
                 "Body mass index in continuous", 
                 "Smoking status (1: started smoking, 0: non-smoker)", 
                 "Systolic blood pressure", 
                 "Diastolic blood pressure")
data_types_bp <- sapply(bloodpressure, class)
percentage_nas_bp <- sapply(bloodpressure, function(x) mean(is.na(x))) * 100
number_of_nas <- sapply(bloodpressure, function(x) sum(is.na(x)))
data_summary_bp <- data.frame(Variable = variable_bp, 
                              Description = description_bp,
                              Data_Type = data_types_bp,
                              NAs_Percent = round(percentage_nas_bp,2))
row.names(data_summary_bp) <- NULL
knitr::kable(data_summary_bp)
```

```{r, include = FALSE}

snps <- snps %>%
  mutate(across(starts_with("rs"), as.numeric),
  sex = as.factor(sex),
  IID = as.character(IID))
str(snps)
```

```{r}
#| label: tbl-desc-snps
#| tbl-cap: The SNPs Dataset Overview
#| tbl-align: center

percentage_nas_snps <- sapply(snps, function(x) mean(is.na(x))) * 100
data_types_snps <- sapply(snps, class)
data_summary_snps <- data.frame(
  Variable = colnames(snps),
  Data_Type = data_types_snps,
  NAs_Percent = round(percentage_nas_snps, 2))
row.names(data_summary_snps) <- NULL
knitr::kable(data_summary_snps)
```

## Study Design

This study used baseline data from the Glasgow Medical Centre Clinical as it shares the similar and small population which likely increase the possibility of Linkage Disequilibrium (LD). Meanwhile LD represents the non-random association of alleles at adjacent loci in a given population and can be affected by numerous factors such as genetic drift and population structure [@Ardlie2002]. This leads to escalate the occurrence of certain genetic variants to be inherited more often. On the other hand, considering this factor, it will impact the outcome of analysis.

The clinical covariates in the blood pressure dataset range from *age*, *sex*, *bmi* and *newsmoke* while the response variables of *sbp* and *dbp* are also stated in this dataset. Then, the summary of the numerical variables in the blood pressure dataset will be given in the table below:

```{r}
#| label: tbl-summary-bloodpressure
#| tbl-cap: The Summary of Numerical Variables in Blood Pressure Dataset 
#| tbl-align: center 
#| echo: false

summary_table_bp <- bloodpressure %>%
  summarise(
    Min_Age = min(age, na.rm = TRUE),
    Max_Age = max(age, na.rm = TRUE),
    Mean_Age = mean(age, na.rm = TRUE),
    Min_BMI = min(bmi, na.rm = TRUE),
    Max_BMI = max(bmi, na.rm = TRUE),
    Mean_BMI = mean(bmi, na.rm = TRUE),
    Min_SBP = min(sbp, na.rm = TRUE),
    Max_SBP = max(sbp, na.rm = TRUE),
    Mean_SBP = mean(sbp, na.rm = TRUE),
    Min_DBP = min(dbp, na.rm = TRUE),
    Max_DBP = max(dbp, na.rm = TRUE),
    Mean_DBP = mean(dbp, na.rm = TRUE)
  )%>%
  pivot_longer(
    cols = everything(),
    names_to = "Variable",
    values_to = "Value"
  ) %>%
  separate(Variable, into = c("Measure", "Variable"), sep = "_") %>%
  pivot_wider(
    names_from = Measure,
    values_from = Value
  ) %>%
  mutate(across(where(is.numeric),~round(.x,2)))
knitr::kable(summary_table_bp)
```

Consequently the genetic factors on SNP dataset are portrayed by 14 identifier SNP with different specific alleles {A, C, G, T}. These variables are treated as the numerical variables where the genotype is encoded as 0, 1, or 2 corresponding to the three possible combinations of alleles {aa, ab, bb}.

To address the research questions, I utilized Multivariate Linear Regression to examine the relationship between the covariates and response variables. Then, particularly for the selection of SNPs variables, I employed residual-based approach and Lasso regularization.

## Data Processing

### Handling with Collinearity

A statistical technique to identify collinearity is by examining the correlation matrix of the predictors. Colinearity occurs when two or more predictor variables are highly correlated with each other. This can create challenges in model building, as it becomes difficult to recognize the individual effects of collinear variables on the response variable [@james2013introduction]. Meanwhile, the Pearson Correlation Coefficient, one of the most commonly used methods for this purpose, assess the strength and direction of the linear relationship between two continuous variables [@james2013introduction].

```{r, include = FALSE}

numeric_vars_bp <- bloodpressure %>% select_if(is.numeric)
cor_matrix_bp <- cor(numeric_vars_bp, use = "complete.obs")
cor_matrix_bp_melt <- melt(cor_matrix_bp)
ggplot(data = cor_matrix_bp_melt, aes(Var1, Var2, fill = value)) +
  geom_tile(color = "white") +
  scale_fill_gradient2(low = "black", high = "red", mid = "white",
                       midpoint = 0, limit = c(-1, 1), space = "Lab",
                       name = "Pearson\nCorrelation") +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 1, 
                                   size = 5, hjust = 1), 
        axis.text.y = element_text(size = 5),
        axis.title.x = element_blank(), 
        axis.title.y = element_blank(), 
        plot.title = element_blank()) +
  coord_fixed() +
  geom_text(aes(Var1, Var2, label = round(value, 2)), color = "black", size = 2)
```

The results of Pearson correlation on blood pressure datasets presented that all covariates exhibited low correlations with each other, suggesting that collinearity was not a concern. However, the clinical covariates showed weak correlations with blood pressure, implying that these variables may have limited predictive power when considered individually. Further analysis is warranted, especially since this heatmap was generated using the initial dataset, which still contained missing values.

```{r}
#| label: fig-corrsnps
#| fig-cap: The Correlation Heatmap of the Initial SNPs Dataset
#| fig-align: center 
#| fig-width: 5 
#| fig-height: 5
#| echo: false

snps_numeric <- snps %>%
  select(-IID, -sex)
cor_matrix_snps <- cor(snps_numeric, use = "complete.obs")
cor_matrix_snps[upper.tri(cor_matrix_snps)] <- NA
cor_matrix_snps_melt <- melt(cor_matrix_snps, na.rm = TRUE)
ggplot(data = cor_matrix_snps_melt, aes(Var2, Var1, fill = value)) +
  geom_tile(color = "white") +
  scale_fill_gradient2(low = "blue", high = "red", mid = "white",
                       midpoint = 0, limit = c(-1, 1), space = "Lab",
                       name = "Pearson Correlation") +
  theme_minimal() + 
  theme(
    axis.text.x = element_text(angle = 90, vjust = 1, hjust = 1, size = 7, color = "black"),
    axis.text.y = element_text(size = 7, color = "black"),
    axis.title.x = element_blank(), 
    axis.title.y = element_blank(),
    panel.grid.major = element_blank(),
    panel.border = element_blank(),
    panel.background = element_blank(),
    axis.ticks = element_blank(),
    legend.position = "bottom",  # Move legend to the bottom
    legend.title = element_text(size = 10),
    legend.text = element_text(size = 8)
  ) +
  coord_fixed() +
  geom_text(aes(Var2, Var1, label = round(value, 2)), color = "black", size = 2) +
  scale_x_discrete(position = "top") +  
  scale_y_discrete(position = "left") +  
  labs(fill = "Pearson Correlation")
```

Meanwhile, based on the correlation heatmap @fig-corrsnps, there were found several SNPs highly correlated - *rs12413409_A*, *rs2681492_G*, and *rs3184504_T* - and showed by the correlation coefficients exceeding 0.95. To address this collinearity issues in the analysis, these SNPs should be removed to increase the model's efficiency, focusing the analysis on the remaining SNPs that are less correlated. The high correlations between these SNPs suggest that they may reside within the same Linkage Disequilibrium (LD) block, indicating that they are likely inherited together due to their physical closeness on the chromosome. LD blocks are part of the genome where SNPs are highly correlated because they are inherited together [@johnson2001haplotype]. By removing SNPs within the same LD block, we can avoid redundancy and ensure the analysis focuses on the most informative genetic variants.

```{r, include = FALSE}

highly_correlated_snp <- which(abs(cor_matrix_snps) > 0.9, arr.ind = TRUE)
highly_correlated_snp <- highly_correlated_snp[highly_correlated_snp[, 1] != highly_correlated_snp[, 2], ]
snps <- snps %>% select(-rs12413409_A, -rs2681492_G, -rs3184504_T)
```

### Handling with Missing Values

In addressing the issue of missing values, I opted to retain all variables since the proportion of missing data for each variable was less than 5%. To handle the missing values, I employed the MissRanger imputation method, which utilizes a Random Forest approach. This method is particularly effective as it can handle multivariate data consisting of continous and categorical variables simultaneously and also preserves the underlying data distribution while providing robust imputations [@stekhoven2011missforest]. To ensure the quality of the imputation, I conducted a thorough comparison of the distributions before and after imputation which allows me to maintain the integrity of the dataset while minimizing the impact of missing data on subsequent analyses.

```{r, include = FALSE}

bloodpressure_to_impute <- bloodpressure %>% select(-IID)
bloodpressure_impute <- missRanger(
  bloodpressure_to_impute, 
  formula = .~.,
  num.trees = 1000, 
  seed = 3)
```

```{r}
#| label: tbl-comparison-imputation
#| tbl-cap: The Comparison of Summary Statistics Before and After Imputation
#| tbl-align: center 
#| echo: false

### Compare the impact of imputation on numerical variables
numeric_bp_to_impute <- bloodpressure_to_impute %>% select_if(is.numeric)
numeric_bp_impute <- bloodpressure_impute %>% select_if(is.numeric)
mean_before_bp <- sapply(numeric_bp_to_impute, function(x) mean(x, na.rm = TRUE))
sd_before_bp <- sapply(numeric_bp_to_impute, function(x) sd(x, na.rm = TRUE))
mean_after_bp <- sapply(numeric_bp_impute, function(x) mean(x, na.rm = TRUE))
sd_after_bp <- sapply(numeric_bp_impute, function(x) sd(x, na.rm = TRUE))
comparison_table_bp <- data.frame(
  Variable = names(mean_before_bp),
  Mean_Before = round(mean_before_bp, 2),
  Mean_After = round(mean_after_bp, 2),
  SD_Before = round(sd_before_bp, 2),
  SD_After = round(sd_after_bp, 2)
)
row.names(comparison_table_bp) <- NULL
knitr::kable(comparison_table_bp)
```

After conducting the imputation on *blood pressure* dataset, it is crucial to verify that the process has preserved the underlying data structure and distribution. As shown in @tbl-comparison-imputation, the mean values remain stable before and after imputation, indicating that the imputation method has effectively filled in the missing data without introducing bias. This consistency suggests that the data's integrity has been successfully maintained. In the case of the SNPs dataset, where missing values are below 1%, the minimal impact of imputation further supports the preservation of the original data distribution.

```{r, include = FALSE}
snps_to_impute <- snps %>% select(-IID)
snps_impute <- missRanger(
  snps_to_impute, 
  formula = .~.,
  num.trees = 1000, 
  seed = 3)
```

```{r, include = FALSE}
bloodpressure_impute$IID <- bloodpressure$IID
snps_impute$IID <- snps$IID
data_combined <- merge(bloodpressure_impute, snps_impute, by = c("IID","sex"))
str(data_combined)
summary(data_combined)
```

\clearpage

### Handling with Outliers

```{r}
#| label: fig-outlierscombined
#| fig-cap: The Outliers Summary of the Combined Data
#| fig-align: center 
#| fig-width: 8 
#| fig-height: 4
#| echo: false

non_snp_vars <- data_combined %>%
  select(-starts_with("rs")) %>%
  select_if(is.numeric) 
covariate_non_snp <- names(non_snp_vars)
covariate_non_snp_plots <- lapply(covariate_non_snp, function(covariate) {
  ggplot(data_combined, aes_string(y = covariate)) +
    geom_boxplot(fill = "white") +
    theme_minimal() +
    labs(title = paste("Boxplot of", covariate), x = "", y = covariate)
})
do.call(grid.arrange, c(covariate_non_snp_plots, ncol = 4))
```

Although these points fell outside the interquartile range, the boxplots indicated that these outliers did not appear to be extreme enough to exert undue influence on the dataset as a whole. Without evidence that these points have high leverage or significant influence on the analysis, their inclusion is warranted. Removing outliers can sometimes lead to a loss of data integrity, especially that these points represent valid observations. By retaining these points, I ensure that the analysis reflects the true variability in the dataset. As the result, the final combined data included 3,808 observations.

# ANALYSIS OF THE DATA

## Data Visualization

```{r}
#| label: fig-linearsbp
#| fig-cap: The Linear Relationship of Clinical Covariates with the SBP Response Variable
#| fig-align: center
#| fig-width: 10 
#| fig-height: 4
#| echo: false

sbp_plots <- list()
clinical_covariates_sbp <- c("age", "bmi", "sex", "newsmoke")
for (predictor in clinical_covariates_sbp) {
  if (is.numeric(data_combined[[predictor]])) {
    p_sbp <- ggplot(data_combined, aes_string(x = predictor, y = "sbp")) +
      geom_point(alpha = 0.5) +
      geom_smooth(method = "lm", se = FALSE, color = "blue") +
      theme_minimal() +
      labs(title = paste("SBP vs", predictor), x = predictor, y = "SBP")
  } 
  else if (is.factor(data_combined[[predictor]])) {
    p_sbp <- ggplot(data_combined, aes_string(x = predictor, y = "sbp", fill = predictor)) +
      geom_violin(trim = FALSE, alpha = 0.7) +
      geom_boxplot(width = 0.1, color = "black", alpha = 0.5) +
      theme_minimal() +
      labs(title = paste("SBP vs", predictor), x = predictor, y = "SBP")
  }
  sbp_plots[[predictor]] <- p_sbp
}
grid.arrange(grobs = sbp_plots, ncol = 4)
```

These observations indicated that among the clinical covariates examined, *age* showed the most evident linear relationship with the *sbp* response variable. The scatter plot demonstrated a positive trend, indicating that as *age* increases, *sbp* tends to rise slightly. Similarly, the relationship between *sex* and *sbp* revealed slight differences in both median and mean *sbp* values between males and females. Males (*sex* 1) exhibit a wider distribution of *sbp* values, suggesting a broader range of blood pressure readings compared to females (*sex* 2). In contrast, the relationship between *bmi* and *sbp* appeared to be relatively flat, as indicated by the nearly horizontal regression line, suggesting no significant effect of *bmi* on *sbp*. Likewise, smoking status (*newsmoke*) did not seem to significantly influence *sbp*, as evidenced by the lack of notable differences in median, mean, or overall distribution in both violin plots.

```{r}
#| label: fig-interactionsbp
#| fig-cap: The Interaction between Each Clinical Covariate and Its Effect on the SBP Response Variable
#| fig-align: center
#| fig-width: 10 
#| fig-height: 5
#| echo: false

plot1 <- ggplot(data_combined, aes(x = bmi, y = sbp, color = factor(newsmoke))) +
  geom_point(alpha = 0.5) +
  geom_smooth(method = "lm", se = FALSE) +
  labs(title = "BMI and Smoking Status on SBP",
       x = "BMI",
       y = "SBP",
       color = "Smoking Status") +
  theme_minimal() + 
    theme(
    plot.title = element_text(size = 9),
    axis.title = element_text(size = 8),
    axis.text = element_text(size = 8),
    legend.title = element_text(size = 7),
    legend.text = element_text(size = 7)
  )
plot2 <- ggplot(data_combined, aes(x = age, y = sbp, color = factor(newsmoke))) +
  geom_point(alpha = 0.5) +
  geom_smooth(method = "lm", se = FALSE) +
  labs(title = "Age and Smoking Status on SBP",
       x = "Age",
       y = "SBP",
       color = "Smoking Status") +
  theme_minimal() + 
    theme(
    plot.title = element_text(size = 9),
    axis.title = element_text(size = 8),
    axis.text = element_text(size = 8),
    legend.title = element_text(size = 7),
    legend.text = element_text(size = 7)
  )
plot3 <- ggplot(data_combined, aes(x = bmi, y = sbp, color = factor(sex))) +
  geom_point(alpha = 0.5) +
  geom_smooth(method = "lm", se = FALSE) +
  labs(title = "BMI and Sex on SBP",
       x = "BMI",
       y = "SBP",
       color = "Sex") +
  theme_minimal() + 
    theme(
    plot.title = element_text(size = 9),
    axis.title = element_text(size = 8),
    axis.text = element_text(size = 8),
    legend.title = element_text(size = 7),
    legend.text = element_text(size = 7)
  )
plot4 <- ggplot(data_combined, aes(x = age, y = sbp, color = factor(sex))) +
  geom_point(alpha = 0.5) +
  geom_smooth(method = "lm", se = FALSE) +
  labs(title = "Age and Sex on SBP",
       x = "Age",
       y = "SBP",
       color = "Sex") +
  theme_minimal() +
    theme(
    plot.title = element_text(size = 9),
    axis.title = element_text(size = 8),
    axis.text = element_text(size = 8),
    legend.title = element_text(size = 7),
    legend.text = element_text(size = 7)
  )
grid.arrange(plot1, plot2, plot3, plot4, ncol = 2)
```

The visual analysis of the interaction between each covariate on the *sbp* response variable revealed no significant differences in slopes across the interaction groups. This suggested that the interaction effects between the covariates (*bmi*, *age* with *sex*, and *newsmoke* variables) and *sbp* were minimal or nonexistent. As a result, it indicated that including interaction terms in the regression model may not be necessary, as they were unlikely to contribute significantly to explaining the variation in *sbp*.

```{r}
#| label: fig-lineardbp
#| fig-cap: The Linear Relationship of Clinical Covariates with the DBP Response Variable
#| fig-align: center
#| fig-width: 10 
#| fig-height: 4
#| echo: false

dbp_plots <- list()
clinical_covariates_dbp <- c("age", "bmi", "sex", "newsmoke")
for (predictor in clinical_covariates_dbp) {
  if (is.numeric(data_combined[[predictor]])) {
    p_dbp <- ggplot(data_combined, aes_string(x = predictor, y = "dbp")) +
      geom_point(alpha = 0.5) +
      geom_smooth(method = "lm", se = FALSE, color = "blue") +
      theme_minimal() +
      labs(title = paste("DBP vs", predictor), x = predictor, y = "DBP")
  } 
  else if (is.factor(data_combined[[predictor]])) {
    p_dbp <- ggplot(data_combined, aes_string(x = predictor, y = "dbp", fill = predictor)) +
      geom_violin(trim = FALSE, alpha = 0.7) +
      geom_boxplot(width = 0.1, color = "black", alpha = 0.5) +
      theme_minimal() +
      labs(title = paste("DBP vs", predictor), x = predictor, y = "DBP")
  }
  dbp_plots[[predictor]] <- p_dbp
}
grid.arrange(grobs = dbp_plots, ncol = 4)
```

\clearpage

Furthermore, the plot @fig-lineardbp revealed that among the clinical covariates examined, *age* demonstrated a slight negative linear relationship with the *dbp* response variable. Meanwhile, the relationship between *sex* and *dbp* resulted in small differences on median and mean *dbp* values on males (*sex* 1) and females (*sex* 2). Notably, females exhibited slightly greater variability in diastolic blood pressure than males as indicated by the broader distribution in the violin plot. Then, the relationship between *bmi* and *dbp* also appeared to follow a slight positive linear trend, indicating the higher *bmi* may be associated with the slightly higher diastolic blood pressure. On the other hand, smoking status did not seem to significantly influence *dbp*, as evidenced with no notable differences in mean and median of *dbp* distribution across different categories of these variables.

```{r}
#| label: fig-interactiondbp
#| fig-cap: The Interaction between Each Clincial Covariate and Its Effect on the DBP Response Variable
#| fig-align: center
#| fig-width: 10 
#| fig-height: 5
#| echo: false

plot5 <- ggplot(data_combined, aes(x = bmi, y = dbp, color = factor(newsmoke))) +
  geom_point(alpha = 0.5) +
  geom_smooth(method = "lm", se = FALSE) +
  labs(title = "BMI and Smoking Status on DBP",
       x = "BMI",
       y = "DBP",
       color = "Smoking Status") +
  theme_minimal() + 
   theme(
    plot.title = element_text(size = 9),
    axis.title = element_text(size = 8),
    axis.text = element_text(size = 8),
    legend.title = element_text(size = 7),
    legend.text = element_text(size = 7)
  )
plot6 <- ggplot(data_combined, aes(x = age, y = dbp, color = factor(newsmoke))) +
  geom_point(alpha = 0.5) +
  geom_smooth(method = "lm", se = FALSE) +
  labs(title = "Age and Smoking Status on DBP",
       x = "Age",
       y = "DBP",
       color = "Smoking Status") +
  theme_minimal() + 
   theme(
    plot.title = element_text(size = 9),
    axis.title = element_text(size = 8),
    axis.text = element_text(size = 8),
    legend.title = element_text(size = 7),
    legend.text = element_text(size = 7)
  )
plot7 <- ggplot(data_combined, aes(x = bmi, y = dbp, color = factor(sex))) +
  geom_point(alpha = 0.5) +
  geom_smooth(method = "lm", se = FALSE) +
  labs(title = "BMI and Sex on DBP",
       x = "BMI",
       y = "DBP",
       color = "Sex") +
  theme_minimal() + 
   theme(
    plot.title = element_text(size = 9),
    axis.title = element_text(size = 8),
    axis.text = element_text(size = 8),
    legend.title = element_text(size = 7),
    legend.text = element_text(size = 7)
  )
plot8 <- ggplot(data_combined, aes(x = age, y = dbp, color = factor(sex))) +
  geom_point(alpha = 0.5) +
  geom_smooth(method = "lm", se = FALSE) +
  labs(title = "Age and Sex on DBP",
       x = "Age",
       y = "DBP",
       color = "Sex") +
  theme_minimal() + 
   theme(
    plot.title = element_text(size = 9),
    axis.title = element_text(size = 8),
    axis.text = element_text(size = 8),
    legend.title = element_text(size = 7),
    legend.text = element_text(size = 7)
  )
grid.arrange(plot5, plot6, plot7, plot8, ncol = 2)
```

Similarly with *sbp* response variable, the interaction plots @fig-interactiondbp revealed that there was no significant interaction effect between each covariate *bmi* and *age* with smoking status and *sex* on the *dbp* response variable. The slopes of the regression lines for different groups (by smoking status or *sex*) were nearly parallel, suggesting that the effect of *bmi* or *age* on *dbp* did not vary substantially across these groups. This indicated that interaction terms between these variables may not be necessary in the regression model.

\clearpage

## Data Modeling

### SBP as The Reponse Variable

#### Relationship between the Clinical Factors with the Response Variable

```{r, include = FALSE}

lm_sbp_clinical <- lm(sbp ~ age + sex + bmi + newsmoke, data = data_combined)
summary(lm_sbp_clinical)
```

```{r}
#| label: fig-forestplotsbp
#| fig-cap: The Forest Plot of Regression Coefficients on SBP Linear Regression
#| fig-align: center
#| fig-width: 10 
#| fig-height: 5
#| echo: false

coefficients_sbp_clinical <- summary(lm_sbp_clinical)$coefficients
conf_intervals_sbp_clinical <- confint(lm_sbp_clinical)
plot_data_sbp_clinical <- data.frame(
  Predictor = rownames(coefficients_sbp_clinical),
  Estimate = coefficients_sbp_clinical[, "Estimate"],
  Lower_CI = conf_intervals_sbp_clinical[, 1],
  Upper_CI = conf_intervals_sbp_clinical[, 2],
  p_value = coefficients_sbp_clinical[, "Pr(>|t|)"]
)
plot_data_sbp_clinical$p_label <- ifelse(plot_data_sbp_clinical$p_value < 0.001, "p close to 0", paste0("p=", round(plot_data_sbp_clinical$p_value, 3)))
ggplot(plot_data_sbp_clinical, aes(x = reorder(Predictor, Estimate), y = Estimate)) +
  geom_pointrange(aes(ymin = Lower_CI, ymax = Upper_CI), color = "blue") +
  coord_flip() +
  theme_minimal() +
  labs(y = "Estimate (95% CI)",
       x = "Covariates") +
  geom_text(aes(label = p_label), vjust = -1, size = 3.5) +
  geom_text(aes(label = paste0("[", round(Lower_CI, 2), ", ", 
                               round(Upper_CI, 2), "]")),
            hjust = 0.5, vjust = 2, size = 3.5)
```

The forest plot displayed in @fig-forestplotsbp highlighted the significant clinical predictors of *sbp* in this model, with *sex*, smoking status, and *age* showing strong associations. This was evidenced by their low p-values, all of which were below the 5% significance level, and their confidence intervals, which were entirely positive. Notably, the relatively narrow confidence intervals for *age* and *sex* suggested a precise estimate of their effect sizes. In contrast, *bmi* did not appear to have a significant impact on *sbp*. Although the confidence interval for *bmi* was narrow, it included zero, indicating that *bmi* did not have a substantial effect on *sbp* in this model.

```{r}
#| label: fig-diagnosticsbp
#| fig-cap: The Diagnostic Plot Assumption of SBP Linear Regression
#| fig-align: center
#| fig-width: 5 
#| fig-height: 5
#| echo: false

par(mfrow = c(2,2))
plot(lm_sbp_clinical)
```

Moving to the model assumptions checking, the diagnostic plots suggested that the *sbp* model fit reasonably well, with most assumptions being met. The Residuals vs Fitted values plot indicated that the variance of the residuals was mostly constant, suggesting that the assumption of homoscedasticity held. Additionally, the Q-Q plot showed that the residuals were approximately normally distributed, further validating the model's assumptions. Moreover, the Residuals vs Leverage plot indicated the absence of any high-leverage points with large residuals, confirming that the decision to retain the outliers in the earlier step was justified.

Then the intial estimated linear regression model of clinical measurement for predicting systolic blood pressure is given by the following equation:

$$                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               \text{sbp} = 132.51 + 0.61 \times \text{age} + 4.10 \times \text{sex}_{\text{female}} + 0.039 \times \text{bmi} + 1.47 \times \text{smokingstatus}_{\text{smoker}}
$$

while age and sex were coded as the categorical variables, as stated earlier.

```{r, include = FALSE}

AIC_sbp <- stepAIC(lm_sbp_clinical, direction = "both", trace = FALSE)
summary(AIC_sbp)
aic_sbp_value <- AIC(AIC_sbp)
BIC_sbp <- stepAIC(lm_sbp_clinical, direction = "both", k = log(nrow(data_combined)), trace = FALSE)
summary(BIC_sbp)
bic_sbp_value <- AIC(BIC_sbp)
```

```{r}
#| label: tbl-aic-bic-sbp
#| tbl-cap: Comparison of AIC & BIC Selected Variables in the SBP - Clinical Covariates Linear Model
#| tbl-align: center 
#| echo: false

model_comparison_sbp <- data.frame(
  Model = c("AIC Selected Model", "BIC Selected Model"),
  AIC = c(aic_sbp_value, bic_sbp_value),
  Variables = c("age, sex, newsmoke", "age, sex")
)
knitr::kable(model_comparison_sbp)
```

Following this, variable selection was performed to exclude the irrelevant variables in the model. In this study, the backward AIC and BIC criterion selection were applied, resulting in a final model with minimized value for these criteria.

The model selection process yielded different results depending on the criterion used. The AIC-selected model, which generally favored more complex models, identified *age*, *sex*, and smoking status (*newsmoke*) as significant predictors of systolic blood pressure (*sbp*). In contrast, the BIC-selected model, which applied a stricter penalty for additional variables, retained only *age* and *sex* as significant predictors, excluding smoking status [@james2013introduction].

While the AIC-selected model provided a slightly better fit to the data as indicated by its lower AIC value, the BIC-selected model was more parsimonious and may be preferable when the goal was to minimize model complexity [@james2013introduction]. Both models consistently highlighted *age* and *sex* as key predictors of *sbp*, but the inclusion of smoking status in the AIC model suggested that it might also play a role, albeit a less robust one.

Given the context of this study, where capturing all potential influences on *sbp* as the response variable is crucial, I opted to retain the AIC-selected model. The significantly lower AIC value justifies the inclusion of smoking status as a predictor, acknowledging that while the effect may be less robust, it is still relevant within the model's framework.

\clearpage

#### Relationship between the Genetic Factors with the Response Variable

```{r, include = FALSE}

lm_sbp_clinical_sgnf <- lm(sbp ~ age + sex + newsmoke, data = data_combined)
summary(lm_sbp_clinical_sgnf)
residuals_sbp_snp <- resid(lm_sbp_clinical_sgnf)
snp_vars <- colnames(data_combined)[grep("rs", colnames(data_combined))]
for (snp in snp_vars) {
  model <- lm(residuals_sbp_snp ~ data_combined[[snp]])
  print(summary(model))
}
snp_sbp_formula <- as.formula(paste("residuals_sbp_snp ~", paste(snp_vars, collapse = " + ")))
snp_sbp_model <- lm(snp_sbp_formula, data = data_combined)
summary_snp_sbp <- summary(snp_sbp_model)

### Check the significant snp
snp_sbp_model_sgnf <- lm(sbp ~ rs2782980_T, data = data_combined)
summary(snp_sbp_model_sgnf)
```

```{r, include = FALSE}

### Generating
coefficients_snp_sbp <- summary_snp_sbp$coefficients
conf_intervals_snp_sbp <- confint(snp_sbp_model)
coefficients_snp_sbp <- summary_snp_sbp$coefficients
conf_intervals_snp_sbp <- confint(snp_sbp_model)
snp_sbp_coefficients <- data.frame(
  SNP = rownames(coefficients_snp_sbp),
  Estimate = coefficients_snp_sbp[, "Estimate"],
  Lower_CI = conf_intervals_snp_sbp[, 1],
  Upper_CI = conf_intervals_snp_sbp[, 2],
  p_value = coefficients_snp_sbp[, "Pr(>|t|)"]
)
print(snp_sbp_coefficients)
```

```{r}
#| label: fig-forestplotsnp-sbp
#| fig-cap: The Forest Plot of SNPs Significance on SBP Response Variable
#| fig-align: center
#| fig-width: 10 
#| fig-height: 5
#| echo: false

ggplot(snp_sbp_coefficients, aes(x = Estimate, y = reorder(SNP, Estimate))) +
  geom_pointrange(aes(xmin = Lower_CI, xmax = Upper_CI), color = "blue") +
  geom_vline(xintercept = 0, linetype = "dashed", color = "red") +
  geom_text(aes(label = paste0("p=", round(p_value, 3))), hjust = -0.3, vjust = -0.5, size = 3, color = "black") +
  theme_minimal() +
  labs(x = "Estimate (95% CI)", 
       y = "SNPs")
```

The analysis of SNPs in relation to systolic blood pressure was employed using residual-based approach derived from the model that includes clinical covariates. Residuals which represented the portion of the response variable not explained by the model predictor, allowed for invesgation of the contribution of SNPs to blood pressure independently of known clinical factors [@james2013introduction]. As illustrated in @fig-forestplotsnp-sbp, the SNP *rs2782980_T* was found to have significant impact on *sbp*, as evidenced by its low p-value and the non-zero confidence interval. This suggested that the SNP *rs2782980_T* may play an important role in influencing *sbp* beyond what was accounted for the clinical covariates alone.

```{r, include = FALSE}

# LASSO REGULARIZATION FOR SNPs VARIABLE SECECTION

## Penalize the Covariates using LASSO Regularization
snp_covariates <- data_combined %>% 
  select(starts_with("rs")) %>%
  mutate(across(everything(), as.numeric))

### Standardize the covariates
data_preproc <- preProcess(snp_covariates, method = c("center", "scale"))
snp_standardized <- predict(data_preproc, snp_covariates)

### Create the model matrix
x_sbp <- as.matrix(snp_standardized)
y_sbp <- data_combined$sbp

### Cross-Validation to find the best lambda
cv_lasso_sbp <- cv.glmnet(x_sbp, y_sbp, alpha = 1)

### Get the best lambda
best_lambda_sbp <- cv_lasso_sbp$lambda.min

### Fit LASSO Model using the best lambda
lasso_model_sbp <- glmnet(x_sbp, y_sbp, alpha = 1, lambda = best_lambda_sbp)

### Extract the coefficients
lasso_coeff_sbp <- coef(lasso_model_sbp)
lasso_coeff_sbp <- as.matrix(lasso_coeff_sbp)
print(lasso_coeff_sbp)

### Identify the significant covariates
significant_vars_sbp <- rownames(lasso_coeff_sbp)[lasso_coeff_sbp != 0]
selected_vars_sbp <- significant_vars_sbp[significant_vars_sbp != "(Intercept)"]
data_selected_sbp <- data_combined %>%
  select(IID, sbp, all_of(selected_vars_sbp))
names(data_selected_sbp)
```

Furthermore, the Lasso regularization also effectively narrowed down the list of the potential SNP to one key variant, *rs2782980_T*, which aligned with the findings from the residual-based approach. Lasso achieved this by performing feature selection through shrinking coefficients to zero thereby excluding less important predictors and providing a more parsimonious model [@james2013introduction]. The use of Lasso helped prevent overfitting by simpliflying the model and ensuring the most significant variables were retained, enhacing the model's robutness and interpretability.

```{r}
#| label: tbl-lasso-snp-sbp
#| tbl-cap: The Results of Lasso Regularization for SNP Selection on SBP
#| tbl-align: center 
#| echo: false

lasso_results_snp_sbp <- data.frame(
  SNP_Identifier = significant_vars_sbp,
  Coefficient = round(lasso_coeff_sbp[lasso_coeff_sbp != 0], 3)
)

# Generate the table
knitr::kable(lasso_results_snp_sbp)
```

### DBP as The Response Variable

#### Relationship between the Clinical Factors with the Response Variable

```{r, include = FALSE}

lm_dbp_clinical <- lm(dbp ~ age + sex + bmi + newsmoke, data = data_combined)
summary(lm_dbp_clinical)
```

```{r}
#| label: fig-forestplotdbp
#| fig-cap: The Forest Plot of Regression Coefficients on DBP Linear Regression
#| fig-align: center
#| fig-width: 10 
#| fig-height: 5
#| echo: false

coefficients_dbp_clinical <- summary(lm_dbp_clinical)$coefficients
conf_intervals_dbp_clinical <- confint(lm_dbp_clinical)
plot_data_dbp_clinical <- data.frame(
  Predictor = rownames(coefficients_dbp_clinical),
  Estimate = coefficients_dbp_clinical[, "Estimate"],
  Lower_CI = conf_intervals_dbp_clinical[, 1],
  Upper_CI = conf_intervals_dbp_clinical[, 2],
  p_value = coefficients_dbp_clinical[, "Pr(>|t|)"]
)
plot_data_dbp_clinical$p_label <- ifelse(plot_data_dbp_clinical$p_value < 0.001, "p close to 0", paste0("p=", round(plot_data_dbp_clinical$p_value, 3)))
ggplot(plot_data_dbp_clinical, aes(x = reorder(Predictor, Estimate), y = Estimate)) +
  geom_pointrange(aes(ymin = Lower_CI, ymax = Upper_CI), color = "blue") +
  coord_flip() +
  theme_minimal() +
  labs(y = "Estimate (95% CI)",
       x = "Predictors") +
  geom_text(aes(label = p_label), vjust = -1, size = 3.5) +
  geom_text(aes(label = paste0("[", round(Lower_CI, 2), ", ", 
                               round(Upper_CI, 2), "]")),
            hjust = 0.5, vjust = 2, size = 3.5)
```

Similar to the analysis conducted on systolic blood pressure, the forest plot for diastolic blood pressure indicated that both *age* and *sex* had a statistically significant impact on *dbp*, as evidenced by confidence intervals that did not include zero and very low p-values. This suggested that *age* and *sex* were reliable predictors of diastolic blood pressure within this model. Conversely, body mass index (*bmi*) and smoking status (*newsmoke*) did not show a significant linear relationship with *dbp*, as indicated by their p-values being greater than 0.05 and confidence intervals that included zero. This lack of significance suggested that, within the context of this model, *bmi* and smoking status did not substantially influence diastolic blood pressure.

```{r}
#| label: fig-diagnosticdbp
#| fig-cap: The Diagnostic Plot Assumption of DBP Linear Regression
#| fig-align: center
#| fig-width: 5 
#| fig-height: 5
#| echo: false

par(mfrow = c(2,2))
plot(lm_dbp_clinical)
```

Given the observations on the @fig-diagnosticdbp, the model assumptions were mostly met then the model was generally appropiate. As for the linearity and homoscedasticity, it was seen that the spread of residuals remained the constant as the fitted values increase on Residuals vs Fitted Values and Scale Location plot. While the residuals appeared to be mostly normally distributed although Q-Q plot indicated some light deviations on the tails. Then, The Residuals vs. Leverage plot suggested that there were a few data points with higher leverage that could be influential, but they did not appear to be extreme enough to unduly bias the model.

The initial estimated linear regression model for predicting diastolic blood pressure based on clinical measurements can be expressed as follows:

$$ \text{dbp} = 108.72 - 0.11 \times \text{age} - 0.72 \times \text{sex}_{\text{female}} + 0.038 \times \text{bmi} + 0.47 \times \text{smokingstatus}_{\text{smoker}}
$$

```{r, include = FALSE}

AIC_dbp <- stepAIC(lm_dbp_clinical, direction = "both", trace = FALSE)
summary(AIC_dbp)
aic_dbp_value <- AIC(AIC_dbp)
BIC_dbp <- stepAIC(lm_dbp_clinical, direction = "both", k = log(nrow(data_combined)), trace = FALSE)
summary(BIC_dbp)
bic_dbp_value <- AIC(BIC_dbp)
```

```{r}
#| label: tbl-aic-bic-dbp
#| tbl-cap: The Comparison of AIC & BIC Selected Variables on DBP Linear Model
#| tbl-align: center 
#| echo: false

model_comparison_dbp <- data.frame(
  Model = c("AIC Selected Model", "BIC Selected Model"),
  AIC = c(aic_dbp_value, bic_dbp_value),
  Variables = c("age, sex", "age, sex")
)
knitr::kable(model_comparison_dbp)
```

Then the result of both the AIC and BIC model selection processes identified *age* and *sex* as significant predictors of diastolic blood pressure (*dbp*). While the AIC-selected model included all four covariates (*age*, *sex*, *bmi*, and smoking status), only *age* and *sex* were statistically significant. The BIC-selected model, correspondingly, retained only *age* and *sex,* aligning with the significant predictors identified by the AIC model. Given that the AIC value was lower than the BIC value, indicating a better balance between model fit and complexity, I prefered the AIC-selected model. This model suggested that while *age* and *sex* were the primary factors influencing *dbp*; *bmi* and smoking status might also contribute in subtle ways, although they did not reach statistical significance.

#### Relationship between Genetic Factors with the Response Variable

```{r, include = FALSE}
lm_dbp_clinical_sgnf <- lm(dbp ~ age + sex, data = data_combined)
summary(lm_dbp_clinical_sgnf)
residuals_dbp_snp <- resid(lm_dbp_clinical_sgnf)
snp_vars <- colnames(data_combined)[grep("rs", colnames(data_combined))]
for (snp in snp_vars) {
  model <- lm(residuals_dbp_snp ~ data_combined[[snp]])
  print(summary(model))
}
snp_dbp_formula <- as.formula(paste("residuals_dbp_snp ~", paste(snp_vars, collapse = " + ")))
snp_dbp_model <- lm(snp_dbp_formula, data = data_combined)
summary_snp_dbp <- summary(snp_dbp_model)
```

```{r, include = FALSE}
coefficients_snp_dbp <- summary_snp_dbp$coefficients
conf_intervals_snp_dbp <- confint(snp_dbp_model)
coefficients_snp_dbp <- summary_snp_dbp$coefficients
conf_intervals_snp_dbp <- confint(snp_dbp_model)
snp_dbp_coefficients <- data.frame(
  SNP = rownames(coefficients_snp_dbp),
  Estimate = coefficients_snp_dbp[, "Estimate"],
  Lower_CI = conf_intervals_snp_dbp[, 1],
  Upper_CI = conf_intervals_snp_dbp[, 2],
  p_value = coefficients_snp_dbp[, "Pr(>|t|)"]
)
print(snp_dbp_coefficients)
```

```{r}
#| label: fig-forestplotsnp-dbp
#| fig-cap: The Forest Plot of SNPs Significance on DBP Response Variable
#| fig-align: center
#| fig-width: 10 
#| fig-height: 5
#| echo: false

ggplot(snp_dbp_coefficients, aes(x = Estimate, y = reorder(SNP, Estimate))) +
  geom_pointrange(aes(xmin = Lower_CI, xmax = Upper_CI), color = "blue") +
  geom_vline(xintercept = 0, linetype = "dashed", color = "red") +
  geom_text(aes(label = paste0("p=", round(p_value, 3))), hjust = -0.3, vjust = -0.5, size = 3, color = "black") +
  theme_minimal() +
  labs(x = "Estimate (95% CI)", 
       y = "SNPs")
```

Similarly as data modeling on *sbp,* I employed the residual-based approach to find the candidate of SNPs associated with diastolic blood pressure. It was shown by the visualization on the @fig-forestplotsnp-dbp, that all p-values were greater than 0.05, suggesting that the associations between the SNPs and *dbp* were not statistically significant. Additionally, the confidence intervals for all SNPs crossed the zero line, further reinforcing the conclusion that these SNPs did not have a meaningful impact on *dbp*. This finding was also consistent with the result of Lasso Regularization where all coefficients were shrunk to zero, indicating that none of the SNPs contributed significantly to *dbp* variability.

```{r, include = FALSE}
## Penalize the Covariates using LASSO Regularization
snp_covariates <- data_combined %>% 
  select(starts_with("rs")) %>%
  mutate(across(everything(), as.numeric))

### Standardize the covariates
data_preproc_dbp <- preProcess(snp_covariates, method = c("center", "scale"))
snp_standardized <- predict(data_preproc_dbp, snp_covariates)

### Create the model matrix
x_dbp <- as.matrix(snp_standardized)
y_dbp <- data_combined$dbp

### Cross-Validation to find the best lambda
cv_lasso_dbp <- cv.glmnet(x_dbp, y_dbp, alpha = 1)

### Get the best lambda
best_lambda_dbp <- cv_lasso_dbp$lambda.min

### Fit LASSO Model using the best lambda
lasso_model_dbp <- glmnet(x_dbp, y_dbp, alpha = 1, lambda = best_lambda_dbp)

### Extract the coefficients
lasso_coeff_dbp <- coef(lasso_model_dbp)
lasso_coeff_dbp <- as.matrix(lasso_coeff_dbp)
print(lasso_coeff_dbp)

### Identify the significant covariates
significant_vars_dbp <- rownames(lasso_coeff_dbp)[lasso_coeff_dbp != 0]
selected_vars_dbp <- significant_vars_dbp[significant_vars_dbp != "(Intercept)"]
data_selected_dbp <- data_combined %>%
  select(IID, dbp, all_of(selected_vars_dbp))
```

```{r, include = FALSE}

if(length(significant_vars_dbp) > 1) {
  lasso_results_snp_dbp <- data.frame(
    SNP_Identifier = significant_vars_dbp,
    Coefficient = round(lasso_coeff_dbp[lasso_coeff_dbp != "(Intercept)"], 3)
  )
} else {
  lasso_results_snp_dbp <- data.frame(
    SNP_Identifier = "No significant SNPs found",
    Coefficient = NA
  )
}
knitr::kable(lasso_results_snp_dbp)
```

# DISCUSSION

## Variation on SBP

The variation of systolic blood pressure explained by clinical measurements in the previous analysis accounted for approximately 5.9% as indicated by $R^2$ in 0.059 with *age*, *sex* and smoking status (*newsmoke*) being significant. While this might seem relatively low, it suggested that these clinical factors collectively contributed a modest but significant portion to the explanation of *sbp* variability in the population studied.

Then on the analysis of correlation of SNPs variable with the *sbp*, the *rs2782980_T* was found to be significantly affecting the response variable. The linear model captured by both factors are given by:

```{r, include = FALSE}
sbp_model_final <- lm(sbp ~ age + sex + newsmoke + rs2782980_T, data = data_combined)
summary(sbp_model_final)
```

$$ 
\text{sbp} = 132.82 + 0.61 \times \text{age} + 4.11 \times \text{sex}_{\text{female}} + 1.43 \times \text{smokingstatus}_{\text{smoker}} + 1.17 \times \text{rs2782980\_T} 
$$

While for the model assumptions, it asserts that there were no apparent pattern or any other problems related to the independence or constant residuals. Moreover, the residuals also followed perfectly the normal distribution. The assumptions also showed no influence points to concern with.

```{r}
#| label: fig-diagnostic-finalmodelsbp
#| fig-cap: The Diagnostic Plot Assumption of Final Model of SBP Linear Regression
#| fig-align: center
#| fig-width: 5 
#| fig-height: 5
#| echo: false

par(mfrow = c(2,2))
plot(sbp_model_final)
```

Furthermore, the final model demonstrated that all variables were positively significant to the systolic blood pressure. For each additional unit increase in the respective variable, while holding all other variables constant, the blood pressure was expected to increase by the amount of the corresponding coefficient. Particularly, for the genetic factor *rs2782980_T*, the presence of each additional T allele was associated with an average increase 1.17 mmHg, assuming all other factors remain constant. Moreover, the model explained approximately 6.05% of the variance of *sbp* which was slightly better than when considering only for the clinical covariates. This also suggested that while the included variables were statistically significant, a large portion of the variability in *sbp* remained unexplained by the these factors.

## Variation on DBP

In case of diastolic blood pressure, genetics factors were found to be not significantly correlated within the dataset. Consequently, the variation of the blood pressure explained by clinical measurement, especially by the significant variables - *age* and *sex* - accounting for only 1.34% of the total variance. This small percentage highlighted the limitations of the study could affect the model's explanatory power.

```{r, include = FALSE}
lm_dbp_clinical_sgnf <- lm(dbp ~ age + sex, data = data_combined)
summary(lm_dbp_clinical_sgnf)
```

$$ \text{dbp} = 110.11 - 0.11 \times \text{age} - 0.72 \times \text{sex}_{\text{female}}
$$

Meanwhile for the model equation, all variables had a negative relationship with the diastolic blood pressure, which contrasts with the relationships observed in the systolic blood pressure. Specifically, the model indicated that for each additional unit of *age* or a change in *sex* (e.g., from male to female), while holding other variables constant, diastolic blood pressure was expected to decrease by the corresponding coefficient.

## Limitation

The study was conducted on a relatively small cohort of patients from the Glasgow Medical Centre, which may limit the potential and capability of the findings to larger and more diverse populations. The small sample size can also reduce the statistical power of the study, potentially leading to the non-detection of significant genetic associations with blood pressure. While the previous GWAS study highlighted the consequence of using large sample size which will the affect the relationship of genetic factors to the response variable [@Manipur2024]. Additionally, since the domain came from the same environment, the participants may share similar genetic factors within Linkage Disequilibrium (LD) blocks, which could reduce the variability in genetic factors observed in the study.

\clearpage

# REFERENCES
